src/Orleans.Runtime/GrainDirectory/MultiClusterRegistration/AsyncManualResetEvent.cs

```diff
@@ -0,0 +1,62 @@
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.GrainDirectory
+{
+    /// <summary>
+    /// An async-compatible manual-reset event.
+    /// </summary>
+    /// <see href="https://github.com/StephenCleary/AsyncEx/blob/5ede2ebad24bb3696fd730de2d7e11cda92bf8dc/src/Nito.AsyncEx.Coordination/AsyncManualResetEvent.cs"/>
+    internal sealed class AsyncManualResetEvent
```

Comment:
This is added to make it easier to port the `GlobalSingleInstanceActivationMaintainer` to `TaskSchedulerAgent`, since it was using a `ManualResetEvent`.